src/Sites/app/Runtime/TabCompleteDetection.tsx

```diff
@@ -194,6 +188,86 @@ export class TabCompleteDetection {
 			this.inputValue.next({ message: newMessage, cursorPosition: newCursorPosition });
 		}, 0);
 	}
+
+	/**
+	 * Removes event listeners from the currentInput.
+	 * Finds and sets the new currentInput by querying the DOM.
+	 * Adds event listeners to the new currentInput.
+	 */
+	private updateInput():void{
+		if(this.currentInput){
+			this.removeInputEventInputListeners(this.currentInput);
+		}
+		this.currentInput = this.getInput();
+
+		if (this.currentInput) {
+			this.addInputEventInputListeners(this.currentInput);
+		}
+	}
+
+	/**
+	 * Removes keyListener and finalizeListener from the given input.
+	 *
+	 * @param input the input to remove listeners from
+	 */
+	private removeInputEventInputListeners(input:HTMLInputElement):void{
+		if (typeof this.keyListener === 'function') {
+			input.removeEventListener('keydown', this.keyListener);
+		}
+
+		if (typeof this.finalizeListener === 'function') {
+			input.removeEventListener('change', this.finalizeListener);
+		}
+	}
+
+	/**
+	 * Adds keyListener and finalizeListener to the given input.
+	 *
+	 * @param input the input to add listeners to
+	 */
+	private addInputEventInputListeners(input:HTMLInputElement):void{
+		if(this.keyListener){
+			input.addEventListener('keydown', this.keyListener, {
+				capture: true
+			});
+		}
+
+		if(this.finalizeListener){
+			input.addEventListener('change', this.finalizeListener, {
+				capture: false
+			});
+		}
+	}
+
+	/**
+	 * Listen for chat component updates to the current user's ban status, and update the input on ban status updates.
+	 */
+	private startListeningForBanStatusChanges():void{
+		let chat = (window as any).twitch.getChat();
+		this.originalChatComponentDidUpdate = chat.componentDidUpdate;
+		chat.componentDidUpdate = (previousProps: ChatComponentProps, previousState:ChatComponentState, snapshot:any) => {
+			if(previousProps.currentUserBannedStatusData?.channel?.self?.banStatus != chat.props.currentUserBannedStatusData?.channel?.self?.banStatus){
+				setTimeout(() => this.updateInput(), 0);
+
+				if (!!this.originalChatComponentDidUpdate && typeof this.originalChatComponentDidUpdate === 'function') {
```

Comment:
Shouldn't the execution of componentDidUpdate be passed on regardless of if the banStatus has changed?

Reply:
Yes, I've updated this.